pkg/orchestration/wiring/wiringutil/svccatentangler/plugin.go

```diff
@@ -109,14 +110,18 @@ func (e *SvcCatEntangler) constructServiceInstance(resource *orch_v1.StateResour
 	}, nil
 }
 
-func (e *SvcCatEntangler) constructResourceContract(resource *orch_v1.StateResource, context *wiringplugin.WiringContext) (wiringplugin.ResourceContract, error) {
+func (e *SvcCatEntangler) constructResourceContract(resource *orch_v1.StateResource, smithResource smith_v1.Resource, context *wiringplugin.WiringContext) (wiringplugin.ResourceContract, error) {
```

Comment:
See my branch https://github.com/atlassian/voyager/compare/svccat-shapes#diff-acbf2382321568395dd65543b8ca0b58R124

You don't know if both shapes should be produced.

Reply:
For now we don't need NewBindableIamAccessible shape yet, I'll remove it. I'll add it back when working on refactoring of IAM role plugin.

Reply:
Actually I don't think `SvcCatEntangler` should have any idea about IamRoles.

Consider the case where we have a new set of providers that produce some sort of `KerberosyTokenThings`. Eventually our code will look like this:

```
	IsEnvVarBindable bool
	IsIAMAccessible  bool
	IsKerberosAccesssible bool
	IsTokenProducing  bool
	IsAnotherShape bool
	IsReallyReallyCool  bool
```

I think @kopper current implementation makes sense - a `SvcCatEntangler` should *always* produce a `BindableEnvVar`. Actually, when we refactor this more, a boolean will not be sufficient - we'll need to force the consumers of `SvcCatEntangler` to provide us the actual functions that can generate (or know about) the data:

```
GetEnvVarData  func (ServiceInstance) BindableEnvVarData
GetBinaryData  func (ServiceInstance) BindableBinaryData
GetSecretData  func (ServiceInstance) BindableSecretData
```

Reply:
Egh replied to the wrong comment